R/measurementCohort.R

```diff
@@ -140,8 +141,27 @@ measurementCohort <- function(cdm,
       cohortCodelist |> dplyr::select("concept_id", "cohort_definition_id"),
       by = "concept_id"
     ) |>
-    dplyr::filter(!is.na(.data$cohort_start_date)) |>
-    dplyr::compute(name = name, temporary = FALSE)
+    dplyr::compute(name = name, temporary = FALSE) |>
+    omopgenerics::newCohortTable(
+      cohortSetRef = cohortSet,
+      cohortCodelistRef = cohortCodelist |> dplyr::collect(),
+      .softValidation = TRUE
+    )
+
+  if (!is.null(valueAsConcept) || !is.null(valueAsNumber)) {
+    cli::cli_inform(c("i" = "Applying measurement requirements."))
+    filterExpr <- getFilterExpression(valueAsConcept, valueAsNumber)
+    cohort <- cohort |>
+      dplyr::filter(!!!filterExpr) |>
+      dplyr::compute(name = name, temporary = FALSE) |>
+      omopgenerics::recordCohortAttrition(reason = "Fulfilling measurement value requirements")
```

Comment:
@nmercadeb I feel like this is a little vague, could it be a bit clearer why records are being dropped?

Reply:
This is when we filter to measurements that recorded a certain outcome/value, but I was not sure whether this should be a separate attrition entrance since maybe t should be included in the "Initial qualifying events"?

Reply:
I think i would not include it